packages/react/dialog/src/Dialog.tsx

```diff
@@ -486,35 +478,51 @@ function getState(open: boolean) {
   return open ? 'open' : 'closed';
 }
 
-const LABEL_WARNING_NAME = 'DialogLabelWarning';
+const TITLE_WARNING_NAME = 'DialogTitleWarning';
 
-const [LabelWarningProvider, useLabelWarningContext] = createContext(LABEL_WARNING_NAME, {
+const [WarningProvider, useWarningContext] = createContext(TITLE_WARNING_NAME, {
   contentName: CONTENT_NAME,
   titleName: TITLE_NAME,
   docsSlug: 'dialog',
 });
 
-type LabelWarningProps = {
+type WarningProps = {
   contentRef: React.RefObject<DialogContentElement>;
 };
 
-const LabelWarning: React.FC<LabelWarningProps> = ({ contentRef }) => {
-  const labelWarningContext = useLabelWarningContext(LABEL_WARNING_NAME);
-
-  const MESSAGE = `\`${labelWarningContext.contentName}\` requires a label for the component to be accessible for screen reader users.
+const TitleWarning: React.FC<WarningProps> = ({ contentRef }) => {
+  const titleWarningContext = useWarningContext(TITLE_WARNING_NAME);
 
-You can label the \`${labelWarningContext.contentName}\` by passing a \`${labelWarningContext.titleName}\` component as a child, which also benefits sighted users by adding visible context to the dialog.
+  const MESSAGE = `\`${titleWarningContext.contentName}\` requires a \`${titleWarningContext.titleName}\` for the component to be accessible for screen reader users.
 
-Alternatively, you can use your own component as a title by assigning it an \`id\` and passing the same value to the \`aria-labelledby\` prop in \`${labelWarningContext.contentName}\`. If the label is confusing or duplicative for sighted users, you can also pass a label directly by using the \`aria-label\` prop.
+If you want to hide the \`${titleWarningContext.titleName}\`, you can wrap it with our VisuallyHidden component.
 
-For more information, see https://radix-ui.com/primitives/docs/components/${labelWarningContext.docsSlug}`;
+For more information, see https://radix-ui.com/primitives/docs/components/${titleWarningContext.docsSlug}`;
 
   React.useEffect(() => {
     const hasLabel =
       contentRef.current?.getAttribute('aria-label') ||
       document.getElementById(contentRef.current?.getAttribute('aria-labelledby')!);
 
-    if (!hasLabel) console.warn(MESSAGE);
+    if (!hasLabel) throw new Error(MESSAGE);
+  }, [MESSAGE, contentRef]);
+
+  return null;
+};
+
+const DESCRIPTION_WARNING_NAME = 'DialogDescriptionWarning';
+
+const DescriptionWarning: React.FC<WarningProps> = ({ contentRef }) => {
+  const descriptionWarningContext = useWarningContext(DESCRIPTION_WARNING_NAME);
+
+  const MESSAGE = `Warning: Missing \`Description\` for {${descriptionWarningContext.contentName}}.`;
+
```

Comment:
Looks like your changes to this message have disappeared? It's missing the instructions you had before for adding `aria-labelledby={undefined}` which I think we still need for users that don't want to render a description.